docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,669 @@
+
+C#:
+
+```csharp
+using MyLibrary.Core;
+```
+F#:
+
+```fsharp
+open MyLibrary.Core
+```
+
+And that's it!
+
+Important takeaways:
+
+* Projects are just folders with files in them.
+* There is a `global.json` file at the top level of your library which needs to names of each project (folder name).
+* A project using another project needs to register a dependency in its `project.json` before it can be used.
+
+## How to test libraries on .NET Core
+
+**Note: You can only use DNX to test at this time**
+
+It's important to be able to test across platforms.  You can do this using [Xunit](http://xunit.github.io/), which is also the testing tool used by .NET Core projects.  Getting a test to work across platforms can be done like this:
+
+1. Create a folder for your test project to live in.  It's best to be consistent with your source folder:
+    
+    ```
+    /src
+       /Library
+    /test
+       /LibraryTests
+    ```
+    
+2. Add a `global.json` file to the root directory of your library.
+
+    ```
+    /src
+       /Library
+    /test
+       /LibraryTests
+    global.json
+    ```
+    
+    And put the names of the src and test folders:
+    
+    ```javascript
+    {
+    	"projects": [
+    		"src", "test"
+    	]
+    }
+    ```
+
+3. Add a new `project.json` file under your test library folder.  Make sure you have the following:
+
+    * A reference to `xunit` version `2.1.0-*`.
+    * A version to `xunit.runner.dnx` version `2.1.0-rc1-*`.
+    * A reference to the library under test (e.g. "Library" if its namespace is `Library`).
+    * A new command called `test` which references `xunit.runner.dnx`.
+
+    An example `project.json` testing the library called "Library" might look like this:
+
+    ```javascript
+    {
+    	"commands": {
+    		"test":"xunit.runner.dnx"
+    	},
+    	"frameworks": {
+    		"dnxcore50":{
+    			"dependencies": {
+    				"Library":"",
+    				"System.Runtime":"4.0.0-rc1-*",
+    				"xunit":"2.1.0",
+    				"xunit.runner.dnx": "2.1.0-rc1-*"
+    			}
+    		}
+    	}
+    }
+    ```
+
+4. Grab the dependencies for the `rc1` version of `xunit.runner.dnx` by opening a command prompt, navigating to the root of your library, and typing the following:
+
+    `$ dnu restore -s "https://myget.org/F/xunit"`
+    
+    Due to the pre-release nature of creating libraries right now, you need to get the test runner for Xunit from a different feed.  This will not be necessary once .NET Core and Xunit align for a true release.
+   
+5. Grab all other dependencies across the project:
+
+    ```$ dnu restore```
+
+6. Navigate to your test project and run tests with `dnx test`:
+
+    ```
+    $ cd path-to-test-project
+    $ dnx test
+    ```
+
+And that's it!  You can now test your library across all platforms using nothing but the command line tools.  To continue testing now that you have everything set up, testing your library is very simple:
```

Comment:
A nit ...

"nothing but the command line tools" sounds like we have a point to prove. How about "... across all platforms using simple command line tools."